src/managers/UpdateManager.ts

```diff
@@ -139,6 +141,24 @@ export class UpdateManager {
     /* tslint:enable:no-floating-promises */
 
     // 2. Update push player with external_user_id
+    const pushUpdatePromise = this.sendPushPlayerUpdate(payload);
+    if (await this.isDeviceIdAvailable()) {
+      await pushUpdatePromise;
+    }
+  }
+
+  // Send REST API payload to update the push player record.
+  // Await until we have a push playerId, which is require to make a PUT call.
+  private async sendPushPlayerUpdate(payload: UpdatePlayerOptions): Promise<void> {
+    let deviceId =  (await Database.getSubscription()).deviceId;
+    if (!deviceId) {
+      await awaitSdkEvent(OneSignal.EVENTS.REGISTERED);
+      deviceId = (await Database.getSubscription()).deviceId;
+    }
+    if (!deviceId) {
```

Comment:
I think this makes more sense nested in the above conditional

Reply:
We have to do a final undefined / null check before calling `OneSignalApiShared.updatePlayer` since it requires a valid instance. The compiler thinks line 156 could still return a nullish value.

I'll wrap the call `OneSignalApiShared.updatePlayer` in an if since the early return here doesn't make that clear.

Reply:
Done